vault/sessions.go

```diff
@@ -123,13 +145,18 @@ func (s *KeyringSessions) Retrieve(profile string) (creds sts.Credentials, err e
 }
 
 // Store stores a sessions for a specific profile, expects the profile to be provided, not the source
-func (s *KeyringSessions) Store(profile string, session sts.Credentials, expires time.Time) error {
+func (s *KeyringSessions) Store(profile string, mfaSerial string, session sts.Credentials) error {
 	bytes, err := json.Marshal(session)
 	if err != nil {
 		return err
 	}
 
-	key := fmt.Sprintf("%s session (%d)", profile, expires.Unix())
+	key := fmt.Sprintf(
+		"session:%s:%s:%d",
+		base64Encoding.EncodeToString([]byte(profile)),
+		base64Encoding.EncodeToString([]byte(mfaSerial)),
+		session.Expiration.Unix(),
```

Comment:
What's the reason to base64 - this seems this handles UTF8 fine? If we're turning the key into a data structure perhaps we should be more intentioned about that rather than using regex? Or then again perhaps we should be hashing profile + mfa  instead.

Reply:
Good questions - I tinkered around with this before settling here. These were my reasons:

- I wasn't sure what, if any, limitations (e.g. size, or included character) the supported keyring implementations might have on the strings used for the item key.
- I realized that encoding the mfa serial in the item key reduced the overall diff because the key is expected to have enough information to find a match in `Retrieve`
- I tried to retain the existing filter/parse mechanisms
- Base64 encoding ensures my separator is not present in the key

But, I share your sentiment that this is a bit hacky, and it would be more straightforward to just JSON encode a struct for the key if there aren't any keyring compatibility concerns with that approach.

Reply:
Hmm, it looks like the File keyring backend uses the key as a filename, so the key needs to be a valid filename. I'll think about this a little more, but suggestions welcome.

Reply:
I ended up just swapping the `:` for a `,`, which should ensure the key produces a valid filename on Linux, MacOS and Windows.

I also changed the KeyringSession fields to accurately express the values and consolidated the parsing logic into the parsing function. Hopefully, that eases some of the cognitive burden in reading and maintaining the session code.